### PR TITLE
rename clear_collection to clear_mutable_collection

### PR DESCRIPTION
**Pull request by kshitij12345**
> We should update the name clear_collection to clear_mutable_collection to make it clear that this won't do anything if tuple or other immutable collection is passed.

Related: https://github.com/Lightning-AI/lightning-thunder/issues/366#issuecomment-2095940166

This PR updates the name to reflect what this symbol does.

### CODE REVIEW

**Review by IvanYashchuk — APPROVED** [review]
Thank you

**Comment by IvanYashchuk** [comment]
Code owner review required
Waiting on code owner review from @carmocca, @lantiga, @mruberry, @robieta, and/or @t-vi.

**Review by mruberry — APPROVED** [review]
Thanks for the ping!

**Pull request merged**
